protocols/virtualfund/virtualfund-single-hop_test.go

```diff
@@ -504,297 +382,121 @@ func TestSingleHopVirtualFund(t *testing.T) {
 
 			updatedObj, err := s.Update(e)
 			updated := updatedObj.(VirtualFundObjective)
 			if err != nil {
 				t.Error(err)
 			}
-			if updated.V.SignedStateForTurnNum[0].HasSignatureForParticipant(alice.role) != true {
-				t.Error(`Objective data not updated as expected`)
+
+			switch my.role {
+			case 0:
+				{
+					if !updated.V.SignedStateForTurnNum[1].HasSignatureForParticipant(p1.role) {
+						t.Error(`Objective data not updated as expected`)
+					}
+				}
+			case 1:
+				{
+					if !updated.V.SignedStateForTurnNum[1].HasSignatureForParticipant(alice.role) {
+						t.Error(`Objective data not updated as expected`)
+					}
+				}
+			case 2:
+				{
+					if !updated.V.SignedStateForTurnNum[1].HasSignatureForParticipant(p1.role) {
+						t.Error(`Objective data not updated as expected`)
+					}
+				}
 			}
 
-			// Part 2: a signature on P1's ledger channel (on her left)
+			// Part 2: a signature on a relevant ledger channel
 			f := protocols.ObjectiveEvent{
 				ObjectiveId: s.Id(),
 			}
 			f.SignedStates = make([]state.SignedState, 0)
-
-			ledger, _ := ledger.CreateTestLedger(aliceAlloc, p1Alloc, &p1.privateKey, 0, big.NewInt(0))
-			ss := signState(ledger.PreFundState(), alice)
-
+			someTurnNum := uint64(99)
+			switch my.role {
+			case 0:
+				{
+					s := ledgerChannelToMyRight.PreFundState().Clone()
+					s.TurnNum = someTurnNum
+					ss = state.NewSignedState(s)
+					_ = ss.Sign(&p1.privateKey)
+				}
+			case 1:
+				{
+					s := ledgerChannelToMyRight.PreFundState().Clone()
+					s.TurnNum = someTurnNum
+					ss = state.NewSignedState(s)
+					_ = ss.Sign(&bob.privateKey)
+				}
+			case 2:
+				{
+					s := ledgerChannelToMyLeft.PreFundState().Clone()
+					s.TurnNum = someTurnNum
+					ss = state.NewSignedState(s)
+					_ = ss.Sign(&p1.privateKey)
+				}
+			}
 			f.SignedStates = append(f.SignedStates, ss)
 
 			updatedObj, err = s.Update(f)
 			updated = updatedObj.(VirtualFundObjective)
 			if err != nil {
 				t.Error(err)
 			}
-			if !updated.ToMyLeft.ledgerChannelAffordsExpectedGuarantees() != true {
-				t.Error(`Objective data not updated as expected`)
+
+			switch my.role {
+			case 0:
+				{
+					if !updated.ToMyRight.Channel.SignedStateForTurnNum[someTurnNum].HasSignatureForParticipant((updated.ToMyRight.Channel.MyIndex + 1) % 2) {
+						t.Error(`Objective data not updated as expected`)
+					}
+				}
+			case 1:
+				{
+					if !updated.ToMyRight.Channel.SignedStateForTurnNum[someTurnNum].HasSignatureForParticipant((updated.ToMyRight.Channel.MyIndex + 1) % 2) {
+						t.Error(`Objective data not updated as expected`)
+					}
+				}
+			case 2:
+				{
+					if !updated.ToMyLeft.Channel.SignedStateForTurnNum[someTurnNum].HasSignatureForParticipant((updated.ToMyLeft.Channel.MyIndex + 1) % 2) {
+						t.Error(`Objective data not updated as expected`)
+					}
+				}
 			}
 
 		}
-
-		t.Run(`New`, testNew)
-		t.Run(`Update`, testUpdate)
 		t.Run(`Crank`, testCrank)
+		t.Run(`Update`, testUpdate)
```

Comment:
Is there a reason we removed the `testNew` test?

Reply:
No reason actually, I just forgot to put it back in! Fixed in [c24e4f8](https://github.com/statechannels/go-nitro/pull/272/commits/c24e4f825b6378f4b9fb0bac035b74cd14221d5a)